Gets our food and drinks content up to date with /tg/, adds Boomer grill

Cenrus opened the pull request:
## Related PRs

https://github.com/tgstation/tgstation/pull/44914
https://github.com/tgstation/tgstation/pull/45633
https://github.com/tgstation/tgstation/pull/46196
https://github.com/tgstation/tgstation/pull/46898
https://github.com/tgstation/tgstation/pull/48050
https://github.com/tgstation/tgstation/pull/46583
https://github.com/tgstation/tgstation/pull/45840
https://github.com/tgstation/tgstation/pull/44322
https://github.com/tgstation/tgstation/pull/49192
https://github.com/tgstation/tgstation/pull/46933
https://github.com/tgstation/tgstation/pull/50622
https://github.com/tgstation/tgstation/pull/46844
https://github.com/tgstation/tgstation/pull/45632
https://github.com/tgstation/tgstation/pull/50065

## Why it's good for the game

Better hamburger sprites, more things for the chef to make. Many thanks to /tg/ for these.

Closes #1674

:cl: Krysonism, trilbyspaceclone, Granpawalton, ArcaneMusic, Sneeker134, SteelSlayer, Fikou, Twaticus, bobubeu
add: More cooking content, ribs, cakes, new burgers.
imageadd: Burger resprites
add: Added new craftable Mob, the Butter Bear.
add: Five new plants!
add: Three new plant reagents!
tweak: Plants with high toxicity are now capable of taking pest damage.
balance: High pest levels are more damaging to plants.
imageadd: Added new plants sprites.
code: Coders can now make plant genes unextractable.
code: Adds the dunkable element. Converts items who were using the dunkable var to instead use this element and removes the dunkable var.
code: Adds a new "DUNKABLE" reagent flag for reagent containers. This determines whether or not the container can have items dunked into it.
add: Added lots of new donuts
add: 3 new dishes cooked with the new plants.
add: Added a nugget box. I'm lovin' it (tm)
del: You can no longer make cherry donuts.
tweak: Tea powder is less toxic and tastes better.
add: snowcones
imageadd: snowiest of snow cones
add: Adds the Grill, which can be powered by coal and wood, increases food nutritional value, taste, and adds grill marks.
add: Adds a new drink, Monkey Energy, works as fuel for the grill and increases your speed if you're a monkey, also adds a new reagent, Char, which works the magic in the grill.
tweak: Coal now turns into CO2 at high temperatures and is now a stack
add: Adds a strawberry ice cream sandwich sprite
add: Strawberry ice cream sandwich crafting recipe
add: Strawberry ice cream sandwich cargo bounty
/:cl:

granpawalton committed:
Improve corpse flower sprite (#44556)

Onule improved the sprites for the corpse flower and I am PRing it
for him since he is away from his typical setup.

granpawalton committed:
Make breakfast foods give a positive moodie before 15 mins (#44914)

Breakfast food now gives a minor but long lasting moodie when they are
eaten within roughly 15 minutes of the shift starting.

Affected foods:

- most egg based foods
- coffee
- bacon
- oatmeal
- orange juice
- milk
- pancakes
- waffles
- donuts
- toasts
- biscuits
- cereal

Buttered toast was moved in the code to where all the other toasts
are. Encourages chef to do more than fry his book and make a silly 10
foot meme sandwich. Coffee from vendors now has an additional use.

Twaticus committed:
snacks cake blackbArry cake coco cake slice coco cake (#46390)

Fikou committed:
grill wrench no longer attacks (#46924)

* fixes grill wrench

* Update code/modules/food_and_drinks/kitchen_machinery/grill.dm

Co-Authored-By: Emmett Gaines <ninjanomnom@gmail.com>

* a

* crr

optimumtact committed:
Merge pull request #47356 from wesoda25/donut

Extends Sprinkle Protection to More Donuts

actioninja committed:
Cleans up plante gene mutability. (#47499)

actioninja committed:
fix crafting ui breaking when knowing desserts (#47615)

Jordie0608 committed:
Merge pull request #48191 from RaveRadbury/meat-donut-balance

Meat donut recipe balance

bob-b-b committed:
Adds strawberry ice cream sandwiches (#50065)

* Adds a new frozen food

A strawberry ice cream sandwich

* Typo in consumable

* adds space in between ice cream

Cenrus committed:
A new pie and four more cakes! (#45633)

Cakes and Pies!
Cooking cat once again saves starving spess mens
About The Pull Request

Adds a new healthy pie!
Adds a few more cakes.

image
Why It's Good For The Game

I see that last added cake and or pie was ages ago! No can do at all tis what I say!
So I added in more cakes for you as a feline fine addition to cooking!
More options with different flavors for are crew to noms with is always a good add I would say all are balanced around hardness of crafting them!
Changelog

cl
add: pie and cakes
imageadd: added some icons and images
/cl

Cenrus committed:
[Ready] More cakes and pies, as well as resprites a few others (#46196)

About The Pull Request

Adds in more tarts - pies
Mime Tart, Lava Coco flow Tart, berry tart
Adds in a few more cakes
Vanilla cake - real this time
Clown cake - Honk!
New book only get able via Dinner-o-mat that teaches you how to makes these
Why It's Good For The Game

More colorful foods and crafting as well as using the locked crafting system - Maybe better looking cakes

Cenrus committed:
replaced book

Cenrus committed:
[READY]Da great American foods update. Many many hamberders, resprites and more! (#48050)

* sprites

*  fiesta

* new bugers, tweaks, bugfix

* ghost burger + boritos

* fiesta fix

* Chickens now drop chicken meat

Cenrus committed:
fix it

Cenrus committed:
BBQ Ribs yeehaw

Cenrus committed:
Butterbear (#46583)

* Created the almighty BUTTERBEAR!!!

* Fixed a subcategory issue.

* Added sound effect to slip, heavily nerfed duration of knockdown

* Changed default name of mob

Trademarked name I guess was a bad idea, but the suggested name is pretty good!

Co-Authored-By: MrDoomBringer <emmanuelssr@gmail.com>

* Balance changes

Bear attack damaged nerfed down to 1. Now is part of the russian faction. Slips now work 100% of the time, but you are unable to slip targets that are already prone.

* Had some indentation errors; oops!

* Less icky

Removed If Else loop for just an if

* Update old name

Co-Authored-By: MrDoomBringer <emmanuelssr@gmail.com>

Cenrus committed:
Merge remote-tracking branch 'BeeStation/master' into cakesandfood

Cenrus committed:
Adds corpse flowers (#44322)

* corpse flower init

* icons

* fixes some bad math

* removes istype helper

* play test changes

* examine text

* .

* found where the conflicts happened

* same

Cenrus committed:
[READY FOR REVIEW]Da botany update: Four new plants and plant traits + trait immutability (#45840)

* bungo

* Adds galaxy thistle and spaceman's trumpet.

* Traits update + new mushroom.

Adds new traits and mutability variable.

* Fixes bungo pit contents

* Bungotoxin lethality is now more in line with other good toxins.

* Adds hair dyeing functionality + lowers enzyme

* flags, plant stat adjustments, runtime fix, spelling and mutations.

* Nerfs toxin content of bungo pits to reduce hypodermic induced lethality

* Name getter fix, adds rarity, better documentation, carnivory tweak

* Bungotoxin has more tells.

* fixies

Cenrus committed:
new plant chems

Cenrus committed:
Merge branch 'corpseflowers' into cakesandfood

Cenrus committed:
Merge pull request #46933 from Krysonism/Somethingforthekids

[READY]Da food update. Donuts! New foods! Nuggbox! Bowl resprite.

Cenrus committed:
fix compile error

Cenrus committed:
Allows some snacks to be dunked in liquids (#46220)

* Makes donuts dunkable

* Moves dunking method to snacks class, rather than just donuts

* Drinks can also be dunked into, but soda cans must be opened

* Making more snacks dunkable, adding proximity check to dunking and cleaning up a little

* Makes fries and fry-likes dunkable

Cenrus committed:
fix jelly_homer

Cenrus committed:
fix more icons

Cenrus committed:
fix donut (and meatball) inhands

Cenrus commented:
PR is ready.

qwertyquerty commented:
Testmerged onto sage

Cenrus commented:
Fixed #1674

PowerfulBacon commented:
Apparently Crafting menu cannot be opened after reading Cooking Desserts 101. It is blue(?|F) screen

Cenrus commented:
Just pushed the fix, should take effect next round.

Cenrus commented:
@qwertyquerty testmerge needs an update for the fix to take place

Cenrus commented:
Wiki updated.

qwertyquerty commented:
@Cenrus check if #1706 is related

Cenrus commented:
@qwertyquerty 1706 isn't a testmerge issue, I just tested it, see the issue post for details.

Cenrus committed:
Merge branch 'master' into cakesandfood

Cenrus committed:
adds the meat clown (#50622)

* adds the meat clown

* funnier

Cenrus commented:
Ready to merj

Fikou commented on code/__DEFINES/reactions.dm:
```diff
@@ -33,7 +33,7 @@
 #define BZ_RESEARCH_SCALE					4
 #define BZ_RESEARCH_MAX_AMOUNT				400
-#define MIASMA_RESEARCH_AMOUNT				160
+#define MIASMA_RESEARCH_AMOUNT				40
```
?????

Cenrus replied:
Was changed as a result of the corpse flower being added in 44322.

Cenrus committed:
46844

arrgh im gonna grill

Cenrus committed:
[Ready] Adds Snow Cones (#45632)

* Update construction.dm

* Update snacks_sandwichtoast.dm

* Update snacks_other.dm

* Create frozen

* Create recipes_frozen

* Update tgstation.dme

* Add files via upload

* .dm!

* .dm!

* Recomplie

* i said .dm! but not snacks_...

* spaces are bad >_<

* We dont have kiwi?

Cenrus committed:
'frozen' food crafting category now works (#46970)

* ice crafting category now works

* un-murder rainbow snowcone

Cenrus committed:
'frozen' food crafting category now works (#46970)

* ice crafting category now works

* un-murder rainbow snowcone

Cenrus committed:
Snowcone polish (#48153)

About The Pull Request

Various aspects of snow cone code made me upset to look at.

    I have fixed some glaring spelling errors, changed descriptions to flow better and reduce redundancy.
    The unused kiwi snowcone is now for a Space Mountain Wind snowcone. (MLG)
    The Mime snow cone had a straw that looked terrible, made it 122 grey. There was also a stray pixel on it, that's gone.
    The honey snowcone now looks less like a clone of lemon, now renamed "amber", it functions for honey as well as apple.
    I removed water as an aspect of the recipe as that makes no sense, snowcones are literally shaved ice and syrup.
    pwrgame snowcones and clown snowcones had their reagents mismatched some reason, that's gone.
    The clown snowcone required clowntears but paid out laughter? That's just cruel. It's now equivalent to the mime snowcone.
    The soda snowcone was clearly intended to be a cola snowcone from the icon, but it was made with sodawater???
    I am literally reliving the trauma of this as I work out all the stuff I changed. RED AND BLUE snowcones were swapped and seriously what the heck.
    Comments state that snow cones inherit SUGAR as a stand in for JUNKFOOD yet some things had SUGAR|JUNKFOOD and that's not cool, removed.
    Fruit salad is made up entirely of citrus fruits, but you would just taste fruits???!!??? I'm dying???!!!??
    Added ice and water tastes to snow cones that were missing it
    Berry snowcones had 10 berry juice for some reason, reduced to 5

Why It's Good For The Game

Misspellings and awkward phrasings are immersion breaking. Incorrect reagent outputs are bad. Making a snowcone sodawater but it looks like cola??? What?

Snow cones should be more enjoyable to engage than this.
Changelog

cl
add: Space Mountain Wind Snowcones
tweak: snow cones do not require water to make
tweak: honey snow cones have more of an amber color
tweak: the mime snow cone has a grey straw so you can actually see it.
tweak: apple snow cones are also amber now
tweak: sodawater snow cones are actually Space Cola snow cones.
tweak: fruit salad snowcone tastes changed
tweak: mixed berry is now just berry
tweak: all snow cones can now taste of ice and water
tweak: clown snow cones recipe changed to require laughter, output 5u of their reagent like almost all other snow cones.
tweak: berry snow cones require and output 5u of berry juice
fix: clown and pwrgame snowcones now contain the appropriate reagents
fix: removed stray pixel on mime snowcone
spellcheck: Misspellings in snowcone titles and descriptions
/cl

I thought about processing ice into shaved ice with the grinder but we also make fried eggs on a table with egg, salt, and pepper so maybe I'll do that another day.

Cenrus committed:
fix compile

Cenrus commented:
Crafting menu broke, one minute

Cenrus committed:
fix crafting

Cenrus committed:
fix donut

Cenrus commented:
Okay, now it's ready. this time for realsies.

Cenrus committed:
damnit

Cenrus committed:
[READY]Da ice cream update 2: On a stick. Adds 5 new ice creams. (#48995)

* blob

* nogga black recipe and fix.

* adds food types and fixes empty ice creams.

Cenrus committed:
reverting

Cenrus committed:
ice cream reverting
